src/main/java/io/split/android/client/FilterBuilder.java

```diff
@@ -0,0 +1,88 @@
+package io.split.android.client;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import io.split.android.client.utils.Logger;
+import io.split.android.client.utils.StringHelper;
+
+public class FilterBuilder {
+    private final List<SplitFilter> filters = new ArrayList<>();
+    private final static String QUERYSTRING_TEMPLATE = "names=[by_name_filters]&prefixes=[by_prefix_filters]";
+
+    static private class SplitFilterComparator implements Comparator<SplitFilter> {
+        @Override
+        public int compare(SplitFilter o1, SplitFilter o2) {
+            return o1.getType().compareTo(o2.getType());
+        }
+    }
+
+    public FilterBuilder addFilter(SplitFilter filter) {
+        filters.add(filter);
+        return this;
+    }
+
+    public String build() {
+
+        if(filters.size() == 0) {
+            return "";
+        }
+
+        StringHelper stringHelper = new StringHelper();
+        StringBuilder queryString = new StringBuilder("");
+        Map<SplitFilter.Type, List<String>> allFilters = new HashMap<>();
```

Comment:
Consider using a set here to dedup values:
`Map<SplitFilter.Type, Set<String>> allFilters = new HashMap<>();`
to avoid passing them to another collection.
Append the list of values via the `addAll` method.

You can also consider using a TreeSet instead of a Hash one, to iterate them in order, or to get an ordered list with the `toArray` method.